crates/sui-types/src/transaction.rs

```diff
@@ -113,6 +122,47 @@ pub enum ObjectArg {
     Receiving(ObjectRef),
 }
 
+#[derive(Debug, PartialEq, Eq, Hash, Clone, Serialize, Deserialize)]
+pub enum Reservation {
+    // Reserve a specific amount of the balance.
+    MaxAmount(u64),
+    // Reserve the entire balance.
+    EntireBalance,
```

Comment:
Nit, I think this variant entry should go first. It is small but it means it will be BCS compatible as is with `Option<u64>`